src/pcre2_compile.c

```diff
@@ -7117,10 +7117,8 @@ for (;; pptr++)
 
         if (lengthptr != NULL)
           {
-          PCRE2_SIZE delta = replicate*(1 + LINK_SIZE);
-          if ((INT64_OR_DOUBLE)replicate*
-                (INT64_OR_DOUBLE)(1 + LINK_SIZE) >
-                  (INT64_OR_DOUBLE)INT_MAX ||
+          INT64_OR_DOUBLE delta = (INT64_OR_DOUBLE)replicate*(INT64_OR_DOUBLE)(1 + LINK_SIZE);
```

Comment:
if a broken int64 cast is being used here, this is likely to make the problem worst by potentially halving the width of delta.

it migth be better IMHO to only cast "replicate" to ensure the multiplication is not being done with plain ints

Reply:
a maybe cleaner way to approach this sort of check pushed (not tested) in the [following branch](https://github.com/carenas/pcre2/tree/intbug).

note that similar logic is used in several other places, so not sure why the static analyzer used didn't pick on those.

Reply:
Your approach is indeed cleaner.
System details: Linux 6.1.3, Clang 14.0.6

Reply:
could you confirm the check is indeed getting removed?, still can't reproduce that with the same compiler as shown by the following simplified code:

```
$ cat i.c
#include <stdio.h>
#include <limits.h>
#include <stdlib.h>

int main(int argc, char *argv[])
{
	int a = atoi(argv[1]);
	
	size_t m;
	       
	m = a * 2;
	if ((long)a * 2 > INT_MAX)
		printf("OVERFLOW\n");
	printf("%zu\n", m);
	return 0;
}
$ clang --version
Debian clang version 14.0.6
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
$ clang -O3 -o i i.c
$ ./i 2147483647
OVERFLOW
18446744073709551614
```

Reply:
That example gives the same result on my system, so this PR was probably a mistake.